spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskMetrics.java

```diff
@@ -88,41 +92,47 @@
 	 */
 	public static final String TASK_EXCEPTION_TAG = "task.exception";
 
-	private Timer.Sample taskSample;
-
 	private LongTaskTimer.Sample longTaskSample;
 
-	private Throwable exception;
+	private Observation observation;
 
 	public void onTaskStartup(TaskExecution taskExecution) {
-		LongTaskTimer longTaskTimer = LongTaskTimer
+		if (this.registry != null) {
+			this.observation = Observation.start(SPRING_CLOUD_TASK_METER, this.registry)
+				.lowCardinalityTag(TASK_EXIT_CODE_TAG, String.valueOf(taskExecution.getExitCode()))
+				.lowCardinalityTag(TASK_STATUS_TAG, STATUS_SUCCESS)
+				.lowCardinalityTag(TASK_EXCEPTION_TAG, "none")
```

Comment:
There is an `observation.error(Throwable)` method, which does the same (and more), can we utilize that instead?

Reply:
Cool, I'll give it a try.